### PR TITLE
Adds 4 CVals to track WebP usage and performance

### PR DESCRIPTION
**Pull request by kaidokert**
This PR adds tracking around AnimatedImage, which is currently only used for WebP decoding.
We are adding 4 metrics
 - Total number of currently decoding WebP images
 - Cumulative total of frames decoded
 - Decoding underruns, when CPU bottleneck is hit
 - Decoding overruns, when timing in decoding pipeline is trying to decode too fast.

b/196887641
b/280381029
b/290130439
b/290060548

### CODE REVIEW

File: cobalt/doc/cvals.md [hunk 1]
```diff
@@ -119,6 +119,18 @@ Here are examples of its usage:
     compatibility violations encountered.
 *   **Count.XHR** - The total number of xhr::XMLHttpRequest in existence
     globally.
+*   **Count.MainWebModule.AnimatedImage.Active** - The total number of currently
+    active animated image decoders. Same image from a single URL source rendered
+    multiple times across the content counts as one decoder.
+*   **Count.MainWebModule.AnimatedImage.DecodedFrame** - Total number of decoded
+    frames across all active image decoders. This number resets only when
+    WebModule gets re-created - e.g. page reload, navigation.
+*   **Count.MainWebModule.AnimatedImage.DecodingUnderruns** - Total number of
```
Comment (bgy36ww) [inline cobalt/doc/cvals.md:128]: What can this data tell us? And how can cert team use this information?

For our use case, if we know each webp is 60 fps. We should just count the DecodedFrame over 1 sec / 60 = # of webp?

So there is nothing such as dropped frames in Webp's world?

File: cobalt/loader/image/animated_webp_image.cc [hunk 2]
```diff
@@ -426,6 +432,25 @@ scoped_refptr<render_tree::Image> AnimatedWebPImage::GetFrameForDebugging(
...
+  AnimatedImageDecodingStats result;
+  if (current_stats.frames_decoded >= last_stats.frames_decoded) {
+    result.frames_decoded =
+        current_stats.frames_decoded - last_stats.frames_decoded;
+    result.frames_underrun =
+        current_stats.frames_underrun - last_stats.frames_underrun;
+    result.frames_overrun =
+        current_stats.frames_overrun - last_stats.frames_overrun;
+  } else {
+    // There was a reset somewhere
```
Comment (bgy36ww) [inline cobalt/loader/image/animated_webp_image.cc:446]: What if the device freezes and no longer decoding frames?

File: cobalt/doc/cvals.md (hunk 1 above)
Comment (kaidokert) [reply cobalt/doc/cvals.md:128]: Starting from the end: Cobalt currently does not drop frames, it stretches/lags the exposure time. E.g frame intended to be displayed for 30 milliseconds can end up 300 milliseconds if CPU was bottlenecked.  See b/290060548

> For our use case, if we know each webp is 60 fps.

If you have 1 webp at fixed 60fps, DecodedFrame should be showing exactly 60, underruns should be 0 and overruns should be 0. If you have 2 of those, DecodedFrames should be 120, and other numbers zero. If CPU is falling behind, you will be getting underruns. If timing in the stack is off, you'll be getting overruns - the latter is less useful for performance checks.

> And how can cert team use this information?

To verify correct playback, inspect the asset you are playing, and verify you are getting all frames decoded in expected timeframe. There should be zero underruns

File: cobalt/loader/image/animated_webp_image.cc (hunk 2 above)
Comment (kaidokert) [reply cobalt/loader/image/animated_webp_image.cc:446]: You would see no increment in numbers over a sampling period.

File: cobalt/doc/cvals.md (hunk 1 above)
Comment (kaidokert) [reply cobalt/doc/cvals.md:128]: Also updated the language here to say frames, not times.

**Review by bgy36ww — APPROVED** [review]
LGTM!

**Pull request merged**